python/ray/tests/test_wait.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+import sys
+import time
+import logging
+import pytest
+
+import ray
+import ray.cluster_utils
+import ray.test_utils
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote
+def f(delay):
+    time.sleep(delay)
+    return
+
+
+def test_wait_returns_1_item_by_default(ray_start_regular):
+    object_ids = [f.remote(0), f.remote(0), f.remote(0), f.remote(0)]
+    ready_ids, remaining_ids = ray.wait(object_ids)
+    assert len(ready_ids) == 1
+    assert len(remaining_ids) == 3
+
+
+def test_wait_with_no_timeout_returns_num_returns_items(ray_start_regular):
+    oids = [ray.put(i) for i in range(10)]
+    (found, rest) = ray.wait(oids, num_returns=2)
+    assert len(found) == 2
+    assert len(rest) == 8
+
+
+def test_wait_with_no_timeout_returns_object_ids_size_if_it_is_less_than_num_returns(
+        ray_start_regular):
+    object_ids = [f.remote(0), f.remote(0), f.remote(0)]
+    ready_ids, remaining_ids = ray.wait(object_ids, num_returns=4)
+    assert len(ready_ids) == 3
+    assert len(remaining_ids) == 0
+
+
+def test_wait_with_timeout_returns_up_to_num_returns_items(ray_start_regular):
+    object_ids = [f.remote(0), f.remote(2)]
```

Comment:
This seems to be the only place that we're using a nonzero argument to `f`. Can we force the second ID to not be ready in a different way (e.g., by using`ray.ObjectID.from_random()`) instead and remove the `time.sleep()` call from `f`? Want to discourage sleeping in tests as much as possible as it's a slippery slope to flaky tests.

Reply:
Pushed a change to use `ray.ObjectID.from_random()`. Appears to work ;)